services/storage/src/simcore_service_storage/s3_client.py

```diff
@@ -387,10 +388,11 @@ async def list_files(
     async def file_exists(self, bucket: S3BucketName, *, s3_object: str) -> bool:
         """Checks if an S3 object exists"""
         # SEE https://www.peterbe.com/plog/fastest-way-to-find-out-if-a-file-exists-in-s3
-        s3_objects, _ = await _list_objects_v2_paginated(
-            self.client, bucket, s3_object, max_total_items=EXPAND_DIR_MAX_ITEM_COUNT
+        response = await self.client.list_objects_v2(
+            Bucket=bucket,
+            Prefix=s3_object,
```

Comment:
So I now checked the history of that file. and **you** actually added that function with the reference above.
Did you now check if it is indeed faster that way?

And also what is the point of going throught the whoel of "Contents" vs len(s3_objects) > 0 ? and why not compare it to 1? there should be only one item right?